crates/evm/traces/src/decoder/precompiles.rs

```diff
@@ -0,0 +1,153 @@
+/// Solidity ABI codec.
+///
+/// Parameter names and types are taken from [evm.codes](https://www.evm.codes/precompiled).
+interface EthereumPrecompiles {
+    /* 0x01 */ function ecrecover(bytes32 hash, uint8 v, uint256 r, uint256 s) returns (address publicAddress);
+    /* 0x02 */ function sha256(bytes data) returns (bytes32 hash);
+    /* 0x03 */ function ripemd(bytes data) returns (bytes20 hash);
+    /* 0x04 */ function identity(bytes data) returns (bytes data);
+    /* 0x05 */ function modexp(uint256 Bsize, uint256 Esize, uint256 Msize, bytes B, bytes E, bytes M) returns (bytes value);
+    /* 0x06 */ function ecadd(uint256 x1, uint256 y1, uint256 x2, uint256 y2) returns (uint256 x, uint256 y);
+    /* 0x07 */ function ecmul(uint256 x1, uint256 y1, uint256 s) returns (uint256 x, uint256 y);
+    /* 0x08 */ function ecpairing(uint256[] x, uint256[] y) returns (bool success);
+    /* 0x09 */ function blake2f(uint32 rounds, uint64[8] h, uint64[16] m, uint64[2] t, bool f) returns (uint64[8] h);
+    /* 0x0a */ function pointEvaluation(bytes32 versionedHash, bytes32 z, bytes32 y, bytes1[48] commitment, bytes1[48] proof) returns (bytes value);
+}
+}
+use EthereumPrecompiles::*;
+
+macro_rules! tri {
+    ($e:expr) => {
+        match $e {
+            Ok(x) => x,
+            Err(_) => return false,
+        }
+    };
+}
+
+/// Tries to decode a precompile call. Returns `true` if successful.
+pub(super) fn decode(trace: &mut CallTrace, _chain_id: u64) -> bool {
+    let [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, x @ 0x01..=0x0a] =
+        trace.address.0 .0
+    else {
+        return false
+    };
+
+    let TraceCallData::Raw(data) = &trace.data else { return false };
+
+    let (signature, args) = match x {
+        0x01 => {
+            let (sig, ecrecoverCall { hash, v, r, s }) = tri!(abi_decode_call(data));
+            (sig, vec![hash.to_string(), v.to_string(), r.to_string(), s.to_string()])
+        }
+        0x02 => (sha256Call::SIGNATURE, vec![data.to_string()]),
+        0x03 => (ripemdCall::SIGNATURE, vec![data.to_string()]),
+        0x04 => (identityCall::SIGNATURE, vec![data.to_string()]),
+        0x05 => (modexpCall::SIGNATURE, tri!(decode_modexp(data))),
+        0x06 => {
+            let (sig, ecaddCall { x1, y1, x2, y2 }) = tri!(abi_decode_call(data));
+            (sig, vec![x1.to_string(), y1.to_string(), x2.to_string(), y2.to_string()])
+        }
+        0x07 => {
+            let (sig, ecmulCall { x1, y1, s }) = tri!(abi_decode_call(data));
+            (sig, vec![x1.to_string(), y1.to_string(), s.to_string()])
+        }
+        0x08 => (ecpairingCall::SIGNATURE, tri!(decode_ecpairing(data))),
+        0x09 => (blake2fCall::SIGNATURE, tri!(decode_blake2f(data))),
+        0x0a => (pointEvaluationCall::SIGNATURE, tri!(decode_kzg(data))),
+        _ => unreachable!(),
+    };
+
+    // TODO: Other chain precompiles
```

Comment:
I'm a bit unsure if we want to implement precompiles for other chains—we had discussed this is a thing we probably don't want to do (albeit we could probably do Optimism's easily?) cc @mattsse

Reply:
https://github.com/foundry-rs/foundry/issues/1078